core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -419,56 +425,66 @@ public boolean accept(CarbonFile file) {
    */
   public static Set<String> getInvalidAndMergedIndexFiles(List<String> indexFiles)
       throws IOException {
     SegmentIndexFileStore indexFileStore = new SegmentIndexFileStore();
     Set<String> mergedAndInvalidIndexFiles = new HashSet<>();
     long lastModifiedTime = 0L;
-    String validIndexFile = null;
+    String validMergeIndexFile = null;
     List<String> mergeIndexFileNames = new ArrayList<>();
     boolean isIndexFilesPresent = false;
     for (String indexFile : indexFiles) {
       if (indexFile.endsWith(CarbonTablePath.INDEX_FILE_EXT)) {
         isIndexFilesPresent = true;
       }
       if (indexFile.endsWith(CarbonTablePath.MERGE_INDEX_FILE_EXT)) {
-        // In case there are more than 1 mergeindex files present, latest one is considered as valid
-        // Ex: In SI table, after small files index merge we will have more than 1 mergeindex files
-        long timeStamp =
+        long indexFileTimeStamp =
             Long.parseLong(CarbonTablePath.DataFileUtil.getTimeStampFromFileName(indexFile));
-        if (timeStamp > lastModifiedTime) {
-          lastModifiedTime = timeStamp;
-          validIndexFile = indexFile;
+        // In case there are more than 1 mergeindex files present, latest one is considered as valid
+        if (indexFileTimeStamp > lastModifiedTime) {
+          lastModifiedTime = indexFileTimeStamp;
+          validMergeIndexFile = indexFile;
         }
         mergeIndexFileNames.add(indexFile);
       }
     }
-    // get the invalid mergeindex files by excluding the valid file.
-    if (mergeIndexFileNames.size() > 1 && validIndexFile != null) {
-      final String validIndexFileName = validIndexFile;
+    //  Possible scenarios where we have >1 merge index file:
+    //  1. SI load when MT has stale index files. As during SI load MT segment file is not updated,
+    //  we directly read from segment directory.
+    //  2. In SI table, after small files index merge(refresh command) we will have more than one
+    //  mergeindex file as we are not deleting old file immediately. If segment file gets deleted,
+    //  then it reads from segment directory.
+    if (mergeIndexFileNames.size() > 1 && validMergeIndexFile != null) {
+      final String validIndexFileName = validMergeIndexFile;
+      // get the invalid mergeindex files by excluding the valid file.
       mergedAndInvalidIndexFiles.addAll(
           mergeIndexFileNames.stream().filter(file -> !file.equalsIgnoreCase(validIndexFileName))
               .collect(Collectors.toSet()));
     }
-    if (isIndexFilesPresent && validIndexFile != null) {
-      indexFileStore.readMergeFile(validIndexFile);
+    // If both index and merge index files are present, read the valid merge index file and add its
+    // mapped index files to the excluding list.
+    // Example: We have xxx.index and xxx.mergeindex files in a segment directory.
+    // Here the merged index file (xxx.index) is considered invalid.
```

Comment:
i think this line is confusing, shouldn't it be like ` Here the index file (xxx.index) is considered invalid.` ?

Reply:
ok, done